core/node/commitment_generator/src/input_generation.rs

```diff
@@ -0,0 +1,40 @@
+use std::fmt;
+
+use zksync_types::{commitment::CommitmentInput, H256};
+
+#[derive(Debug)]
+pub struct ValidiumInputGenerator;
+
+#[derive(Debug)]
+pub struct RollupInputGenerator;
+
+pub trait InputGenerator: 'static + fmt::Debug + Send + Sync {
```

Comment:
Having some documentation explaining the purpose, e.g. what kind of input is being calculated here, and why do we both accept and return input (e.g. based on the signature alone it feels like input post-processor, idk), would be appreciated :)

Reply:
i figured there were 2 ways to go about it:
1. take the input generated and post process it
2. take the raw data and generate the input

i think in the long run we likely will want to do something closer to (2) but i wasn't sure if this needed to be done for the current use case. definitely will add some documentation though